### PR TITLE
fix: Limit IAM permissions for EKS

### PR DESCRIPTION
**Pull request by BoboBananas**
## Summary

Review and limit IAM permissions for EKS Audit Logs Terraform template to adhere to least privilege
## How did you test this change?

https://lacework.atlassian.net/wiki/spaces/ENG/pages/2060877955/EKS+K8s+Audit+Ingestion+Design+Document#Data-Flow - Trimmed permissions using this to understand functionality of the different resources
Tested in dev3
Created resources in devtest-sandbox account
Tested cases with and without debugging permissions and validated no issues

## Issue
https://lacework.atlassian.net/browse/RAIN-58162

### CODE REVIEW

**Review by afiune — COMMENTED** [review]
Could we add an example to the `examples/` folder to show and test enabling the debug statements?

Here is an example https://github.com/lacework/terraform-aws-eks-audit-log/pull/22

Make sure to add it to to the test cases list https://github.com/lacework/terraform-aws-eks-audit-log/blob/main/scripts/ci_tests.sh#L11-L18

File: examples/allow_debugging_permissions/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+| `cluster_names`               | A set of cluster names, to integrate with. Defaults to [] if `no_cw_subscription_filter` is set to `true` | `set(string)`  |
+| `allow_debugging_permissions` | A boolean that can enable debugging permissions                                                           | `boolean`      |
+
+## Sample Code
+
+```terraform
+provider "lacework" {}
+
+module "aws_eks_audit_log" {
+  source = "../.."
```
Comment (afiune) [inline examples/allow_debugging_permissions/README.md]: Update this to use source and version from the Terraform registry instead.

Look at other README's in the example folder.

**Review by afiune — APPROVED** [review]
Nice work!!!!

**Review by rmoles — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**